Update SQL documentation & small typo fix api-877 (#1060)
hazelcast/hazelcast-nodejs-client: DOCUMENTATION.md

The SQL service provided by Hazelcast Node.js client allows you to run SQL queries.

You can use SQL to query data in maps, Kafka topics, or a variety of file systems. Results can be sent directly to the client
or inserted into maps or Kafka topics. For streaming queries, you can submit them to a cluster as jobs to run in the background.

> **WARNING: The SQL feature have become stable in 5.0 versions of the client and the server. In order a client and a server to
> be fully compatible with each other, their major versions must be the same.**

> > **WARNING:  To use the SQL feature, the Jet engine must be enabled on the members and the `hazelcast-sql` module must be in the

Reviewer: ```suggestion
> **WARNING: To use the SQL feature, the Jet engine must be enabled on the members and the `hazelcast-sql` module must be in the
```

> > classpath  of the members. If you are using the CLI, Docker image, or distributions to start Hazelcast members, then you don't

Reviewer: ```suggestion
> classpath of the members. If you are using the CLI, Docker image, or distributions to start Hazelcast members, then you don't
```

> > need to  do anything, as the above preconditions are already satisfied for such members.**

Reviewer: ```suggestion
> need to do anything, as the above preconditions are already satisfied for such members.**
```

>

> > **However, if you are using Hazelcast members in the embedded mode, or receiving errors saying that

Reviewer: Is the `**` before `However` necessary?

Author: Yeah it was necessary. Since there is a newline in between these

SELECT * FROM someMap WHERE this = CAST(? AS INTEGER)
```

#### 8.7.9.2. An Example of Casting

Since Node.js client's number type is like a double, to compare with an integer-like column, you need to use casting.

> Alternatively, you can use `Long` objects to compare with columns with `BIGINT` type.

Reviewer: ```suggestion
Alternatively, you can use `Long` objects to compare columns with `BIGINT` type.
```
Maybe this a bit more clear, WDYT?

Author: I went with a different one:

Alternatively, you can use `Long` objects where you have a column with `BIGINT` type, because `Long` objects correspond to the `BIGINT` type.

* No support for the `CREATE INDEX` statement. To create indexes for maps in Hazelcast, see `addIndex` method of map in
  [API documentation](http://hazelcast.github.io/hazelcast-nodejs-client/api/current/docs/).
* No support for the JSON type. You can’t use functions such as `JSON_VALUE` or `JSON_QUERY`.
* Limited support for joins. See [Join Tables](https://docs.hazelcast.com/hazelcast/latest/sql/select#join-tables).
* No support for window functions. You cannot group or aggregate results in streaming queries.

### 8.7.12. Improving the Performance of SQL Queries

You can improve the performance of queries over maps by indexing map entries. To find out more about indexing map entries, see
`addIndex` method of map in [API documentation](http://hazelcast.github.io/hazelcast-nodejs-client/api/current/docs/).

<!-- markdownlint-disable-next-line MD013 -->
If you find that your queries lead to out of memory exceptions (OOME), consider decreasing the value of the Jet engine’s [max-processor-accumulated-records](https://docs.hazelcast.com/hazelcast/latest/configuration/jet-configuration#list-of-configuration-options)
option.

### 8.7.13. More Information

> Please refer to [Hazelcast SQL docs](https://docs.hazelcast.com/hazelcast/5.0/sql/sql-overview) for more information.

Reviewer: ```suggestion
Please refer to [Hazelcast SQL docs](https://docs.hazelcast.com/hazelcast/latest/sql/sql-overview) for more information.
```
